File: Darwin/YABA/Core/Util/SortType.swift
//
//  SortType.swift
//  YABA
//
//  Created by Ali Taha on 3.05.2025.
//

import Foundation
import SwiftUI

enum SortType: Int, Hashable, CaseIterable {
    case createdAt, editedAt, label, custom
    
    func getUITitle() -> LocalizedStringKey {
        return switch self {
        case .createdAt: LocalizedStringKey("Sort Created At")
        case .editedAt: LocalizedStringKey("Sort Edited At")
        case .label: LocalizedStringKey("Sort Label")
        case .custom: LocalizedStringKey("Sort Custom")
        }
    }
    
    func getUIIconName() -> String {
        return switch self {
        case .createdAt: "clock-04"
        case .editedAt: "edit-02"
        case .label: "sorting-a-z-02"
        case .custom: "custom-field"
        }
    }
}

File: Compose/YABA/app/src/main/java/dev/subfly/yaba/core/model/utils/SortType.kt
package dev.subfly.yaba.core.model.utils

import kotlinx.serialization.Serializable

@Serializable
enum class SortType {
    CREATED_AT,
    EDITED_AT,
    LABEL,
}

fun SortType.uiIconName(): String =
    when (this) {
        SortType.CREATED_AT -> "clock-04"
        SortType.EDITED_AT -> "edit-02"
        SortType.LABEL -> "sorting-a-z-02"
    }
